helix-term/src/commands/lsp.rs

```diff
@@ -705,6 +705,11 @@ fn goto_impl(
             editor.set_error("No definition found.");
         }
         _locations => {
+            for i in (1..locations.len()).rev() {
+                if locations.as_slice()[..i].contains(&locations[i]) {
+                    locations.remove(i);
+                }
+            }
```

Comment:
This is very inefficient. Can't we sort then `dedup_by`?

Reply:
We would have to sort `locations` for that, but Ord trait is not implemented for `lsp_types::Location`.

Reply:
Could we make it a `BTreeSet` instead ? Duplicates would simply disappear. The output order would change though, and LSPs often put the definition in the first or last place, loosing this is not a good solution :/